mlir/lib/Transforms/lab4/sadikov_damir/SadikovMLIRDepthPass.cpp

```diff
@@ -0,0 +1,59 @@
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/Region.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+#include <map>
+using namespace mlir;
+
+namespace {
+class SadikovMLIRDepthPass
+    : public PassWrapper<SadikovMLIRDepthPass, OperationPass<func::FuncOp>> {
+public:
+  StringRef getArgument() const final { return "sadikov-mlir-depth-pass"; }
+  StringRef getDescription() const final {
+    return "sadikov-mlir-depth-pass : pass that counts the max depth of region "
+           "nests in the function";
+  }
+  void runOnOperation() override {
+    Operation *func = getOperation();
+    func->setAttr("depth_of_region",
```

Comment:
Why depth **of region**?

Reply:
Just accidentally named it in wrong way. I think depth_of_func is more correct. Renamed it like this

Reply:
Ok